### DISCUSSION TITLE
虚拟列表Virtual List 和 Table渲染问题

### DISCUSSION THREAD

**Discussion by GeordiIGT** [ant-design/ant-design#36153] [Q&A]
现在项目组中有个需求需要开发一个grid组件，容纳10万条数据，支持层级关系，层级之间可以展开折叠，要有列排序（sorting）和过滤（filtering）功能，不使用任何分页，渲染要尽量流畅包括使用侧边滚动条的过程。
![antd_00](https://user-images.githubusercontent.com/55938406/175188920-46187d6c-8d62-4e46-abf2-e8b75bcf0e55.jpg)
用antd-table可以实现初步的列排序过滤，以及层级展开闭合的功能，但是10万条数据不分页的性能不能满足要求，数据到1万的时候排序刷新或者展开折叠操作就可感知的慢了很多，10万条数据的时候页面任何操作都已经非常卡顿了。

![image](https://user-images.githubusercontent.com/55938406/175189139-d0a67718-ddd5-4565-8be0-9b35305eb06d.png)

我看了一下虚拟列表实例（https://ant.design/components/table-cn/#components-table-demo-virtual-list）中，更新渲染速度优化的很好，数据渲染使用的Grid组件进行实现的，我想问如何将虚拟列表应用在原生table 组件上？用table上的数据结构，通过添加children属性数组值，用来保持table数据中的子与父的层级关系，并同时拥有虚拟列表对于大量数据提速渲染的优越性，可以实现吗？

**Comment by huntdream** [comment]
你要实现的是一个支持虚拟列表的树形表格吗？

**Reply by GeordiIGT** [reply]
是的
1. 有树形结构，可以闭合展开
2. 数据不需要汇总，按所给数据显示即可
3. 在10万条数据不分页的情况下能够流畅访问排序，过滤等功能

**Comment by zengpeiceng** [comment]
我也在做这个需求，请问实现了吗

**Comment by Veveue** [comment]
`ali-react-table`
